controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				spobj.Status.Phase = ocsv1.StorageProfilePhaseFailed
```

Comment:
if the object is not of time `ocsv1.StorageProfile` how can we set the status?

Reply:
But the object is of type `ocsv1.StorageProfile`. Can you clarify?

Reply:
But the object is of type `ocsv1.StorageProfile`. The `StorageProfile` resource is being `watch`ed by `StorageCluster`.

Reply:
@Madhu-1 I agree with @raaizik on this. If we run this code on a non-sp even, the code should blow as it means that somebody made a coding error not a runtime error. Blowing the code up will make it easier to focus on the fact that you are trying to use a handler that was designed for an SP object on a watch that watches another type of object.

Reply:
This check will be hit when the typecasting doesn't happen right? I am just going by code reading.

Reply:
Ok, I understand now. I expected that a type assertion without the `ok` params will be used. That will guarantee that the process panics if the assertions on the code are not correct. In this case, I think that the `ok` param and the `if` are unnecessary and even harmful. Mainly because we want the code to panic if someone has used the watch on the wrong type. What we do not want is for the code to continue to run with no side effect.

@raaizik Can we have the change implemented?

Reply:
Done